pandas/core/internals/array_manager.py

```diff
@@ -794,7 +794,14 @@ def column_arrays(self) -> list[ArrayLike]:
         """
         Used in the JSON C code to access column arrays.
         """
-        return self.arrays
+
+        def convert_array(arr: ArrayLike) -> ArrayLike:
```

Comment:
extract_array ?

Reply:
That doesn't work to convert the `ExtensionArray` to a numpy array:
```python
In [1]: import pandas as pd

In [2]: pd.__version__
Out[2]: '1.4.0.dev0+1440.g5ee2c0229b.dirty'

In [3]: from pandas.core.construction import extract_array

In [4]: a=pd.array([1,2,3])

In [5]: a
Out[5]:
<IntegerArray>
[1, 2, 3]
Length: 3, dtype: Int64

In [6]: extract_array(a, True)
Out[6]:
<IntegerArray>
[1, 2, 3]
Length: 3, dtype: Int64
```

That may be a bug in the implementation of `extract_array()`  .  Should I fix that and use it instead?

Reply:
> That may be a bug in the implementation of extract_array() . Should I fix that and use it instead?

No, that is behaving as intended.

Reply:
> > That may be a bug in the implementation of extract_array() . Should I fix that and use it instead?
> 
> No, that is behaving as intended.

OK, the docs aren't clear, because the parameter `extract_numpy=True` then only works for numpy backed extension arrays, otherwise it is ignored for other EA types (e.g., `Categorical`)

Reply:
> numpy backed extension arrays

yep, that is poor wording on the 'obj' arg; the 'extract_numpy' part of the docstring gets it right.  ill write myself a note to fix the wording.

Reply:
why isn't `np.asarray` the right answer here?

Reply:
> why isn't np.asarray the right answer here?

That would work, yes.